src/goto-symex/symex_target_equation.h

```diff
@@ -320,6 +323,9 @@ class symex_target_equationt:public symex_targett
   // for enforcing sharing in the expressions stored
   merge_irept merge_irep;
   void merge_ireps(SSA_stept &SSA_step);
+
+private:
+  const namespacet &ns;
```

Comment:
This has been removed in e8eec2c8a8 so I'm a bit hesitant seeing it re-introduced. @karkhaz would you recall why it had to be removed?

Reply:
There's a possibility that the namespace that this reference points to would go out of scope if we save this `symex_target_equationt` and then resume it later. For some classes (e.g. `goto_symex_statet`), my solution to this was to [make the copy constructor private](https://github.com/diffblue/cbmc/blob/develop/src/goto-symex/goto_symex_state.h#L380) and [force clients to pass an equation explicitly to the copy constructor](https://github.com/diffblue/cbmc/blob/develop/src/goto-symex/goto_symex_state.h#L38).

I think the reason we decided not to do this for the `ns` member of `symex_target_equationt` was that there were no users of `ns` in the class (or maybe just the `output` method?) so we just decided to delete the member. If you would like to have a namespace reference again, I would suggest that you follow the approach I used in those links I posted above.

Reply:
I suppose if I use `format` I can drop this back out anyway

Reply:
That would be my preferred solution, yes.

Reply:
Means we can't output the `SSA_stept` - so I'd prefer to keep it unless there is an equivalent way to work around that?

Reply:
I'd see two options: make the `output`  function use `format` instead of `from_expr` and remove the `namespacet` parameter or introduce an `output_generic` function that does this while leaving the existing `output` function intact. In a way, such back-end code shouldn't rely on language specifics anyway and thus shouldn't be doing language-specific output either.